### PR TITLE
Feature/mui v5

### PR DESCRIPTION
**Pull request by bkarambe**
#### Changes proposed in this Pull Request:
- Added mui v5 dependencies for all the templated

#### To Test:
- Create template projects using command 
- Update the brightlayer-ui node modules after migration to v5
- npx create-react-app myapp --template file:path

### CODE REVIEW

**Comment by jeffvg** [comment]
looks like all the ```<hidden>``` components are deprecated and should use paper or box in all templates. 

Not 100% sure but the implementation of useStyles makeStyles theme needs updated to 
```const useStyles = makeStyles((theme: Theme) =>``` and not ```const useStyles = makeStyles((theme) =>```

@daileytj

File: authentication-typescript/template.json [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+            "@mui/styles": "^5.5.0",
+            "@emotion/react": "^11.8.2",
+            "@emotion/styled": "^11.8.1",
             "@brightlayer-ui/react-auth-workflow": "^2.4.0",
-            "@brightlayer-ui/react-themes": "^6.1.0",
-            "@brightlayer-ui/react-components": "^5.3.0",
-            "@brightlayer-ui/icons-mui": "^2.4.1",
+            "@brightlayer-ui/react-themes": "^6.1.1",
+            "@brightlayer-ui/react-components": "^5.4.0",
+            "@brightlayer-ui/icons-mui": "^2.8.1",
```
Comment (daileytj) [inline authentication-typescript/template.json:12]: You're still using Material-UI v4 based packages. Let's use the latest alpha @brightlayer-ui packages.

File: authentication-typescript/template/src/pages/home.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export const HomePage = (): JSX.Element => {
     const theme = useTheme();
     const classes = useStyles(theme);
     const { setDrawerOpen } = useDrawer();
-    const xs = useMediaQuery(theme.breakpoints.down('xs'));
+    const xs = useMediaQuery(theme.breakpoints.down('sm'));
```
Comment (daileytj) [inline authentication-typescript/template/src/pages/home.tsx:67]: rename this variable to reflect what it's actually doing. If it's no longer responding at xs let's not call it xs.

File: authentication/template.json [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+            "@mui/styles": "^5.5.0",
+            "@emotion/react": "^11.8.2",
+            "@emotion/styled": "^11.8.1",
             "@brightlayer-ui/react-auth-workflow": "^2.4.0",
-            "@brightlayer-ui/react-themes": "^6.1.0",
-            "@brightlayer-ui/react-components": "^5.3.0",
-            "@brightlayer-ui/icons-mui": "^2.4.1",
+            "@brightlayer-ui/react-themes": "^6.1.1",
+            "@brightlayer-ui/react-components": "^5.4.0",
+            "@brightlayer-ui/icons-mui": "^2.8.1",
```
Comment (daileytj) [inline authentication/template.json:12]: use latest @brightlayer-ui alpha packages

File: authentication/template/src/App.test.jsx [hunk 4]
```diff
@@ -14,16 +14,18 @@ jest.mock('@brightlayer-ui/react-auth-shared', () => ({
...
-                value={{
-                    drawerOpen: true,
-                    setDrawerOpen: jest.fn(),
-                }}
-            >
-                <HomePage />
-            </DrawerContext.Provider>
-        </ThemeProvider>
+        <StyledEngineProvider injectFirst>
+            <ThemeProvider theme={createTheme(adaptV4Theme(BLUIThemes.blue))}>
```
Comment (daileytj) [inline authentication/template/src/App.test.jsx:18]: I don't think we need the v4 theme adapter if you use the latest alpha theme that was built for MUIv5

File: routing-typescript/template.json [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
-            "@brightlayer-ui/react-components": "^5.3.0",
-            "@brightlayer-ui/icons-mui": "^2.4.1",
+            "@mui/material": "^5.5.0",
+            "@mui/icons-material": "^5.5.0",
+            "@mui/styles": "^5.5.0",
+            "@emotion/react": "^11.8.2",
+            "@emotion/styled": "^11.8.1",
+            "@brightlayer-ui/react-themes": "^6.1.1",
+            "@brightlayer-ui/react-components": "^5.4.0",
+            "@brightlayer-ui/icons-mui": "^2.8.1",
```
Comment (daileytj) [inline routing-typescript/template.json:11]: 👀

File: routing/template/src/pages/home.jsx [hunk 6]
```diff
@@ -57,7 +57,7 @@ export const HomePage = () => {
     const theme = useTheme();
     const classes = useStyles(theme);
     const { setDrawerOpen } = useDrawer();
-    const xs = useMediaQuery(theme.breakpoints.down('xs'));
+    const xs = useMediaQuery(theme.breakpoints.down('sm'));
```
Comment (daileytj) [inline routing/template/src/pages/home.jsx:60]: 👀

File: routing/template/src/pages/home.jsx [hunk 7]
```diff
@@ -95,7 +95,7 @@ export const HomePage = () => {
                             Edit <strong>src/pages/home.jsx</strong> and save to reload.
                         </Typography>
                     </div>
-                    <Hidden xsDown>
+                    <Hidden smDown>
```
Comment (daileytj) [inline routing/template/src/pages/home.jsx:98]: deprecated 👀

**Review by daileytj — CHANGES_REQUESTED** [review]
@bkarambe, Lots of duplicate changes need to be made here. I commented on the first few and then threw some 👀's on the remaining. I'm sure you'll see the pattern of what you need to update

**Comment by daileytj** [comment]
You'll also need to update the CHANGELOG.md for each template as well as update the package version number. Probably best to make this a major update.

**Comment by daileytj** [comment]
> looks like all the `<hidden>` components are deprecated and should use paper or box in all templates.
> 
> Not 100% sure but the implementation of useStyles makeStyles theme needs updated to `const useStyles = makeStyles((theme: Theme) =>` and not `const useStyles = makeStyles((theme) =>`
> 
> @daileytj

If it's typescript, the type will need to be explicitly set as it is not inferred for the theme. If it's a javascript template, we won't need it because there are no types/interfaces.

**Comment by ektasawant** [comment]
1. Padding should be 16px as per current design
Screenshot 2022-03-17 at 12 07 42 PM
2. DOM nesting error
Screenshot 2022-03-17 at 11 54 34 AM
3. dot appears in next line since <p> tag is used. In our current version we are using <span>.
Screenshot 2022-03-17 at 11 54 39 AM

**Comment by ektasawant** [comment]
Update Package.json with following:
1. @brightlayer-ui/react-themes@7.0.0-alpha.1
2. @brightlayer-ui/react-auth-workflow@3.0.0-alpha.0

**Comment by ektasawant** [comment]
> 1. Padding should be 16px as per current design
> 
> Screenshot 2022-03-17 at 12 07 42 PM
> 
> 2. DOM nesting error
> Screenshot 2022-03-17 at 11 54 34 AM
> 
> 3. dot appears in next line since
> tag is used. In our current version we are using . Screenshot 2022-03-17 at 11 54 39 AM

Tested above fixes. Working fine 👍 .

**Review by ektasawant — APPROVED** [review]

**Comment by emclaug2** [comment]
I ran through these templates yesterday; looks good to me.  If you're happy with the changes @daileytj this should be good to go.

**Review by emclaug2 — APPROVED** [review]

**Pull request merged**